File: WorldFeed/Server/src/WorldFeed.Logic/TweetWithSearchMetadata.cs
namespace WorldFeed.Logic
{
    using WorldFeed.Common.Models;
    using WorldFeed.Common.Public;
    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;
    using WorldFeed.Common.Settings;

    public class TweetWithSearchMetadata : Tweet, ITweetWithSearchMetadata
    {
        private readonly ITweetWithSearchMetadataDTO tweetWithSearchMetadataDTO;

        public TweetWithSearchMetadata(ITweetWithSearchMetadataDTO tweetDTO, TweetMode? tweetMode, ITwitterClient client)
            : base(tweetDTO, tweetMode, client)
        {
            this.tweetWithSearchMetadataDTO = tweetDTO;
        }

        public ITweetFromSearchMetadata SearchMetadata => this.tweetWithSearchMetadataDTO.TweetFromSearchMetadata;
    }
}

File: Sharebook/Client/src/app/logic/TweetWithSearchMetadata.ts
import {Tweet} from "../core/Core/Models/Tweet";
import {ITweetWithSearchMetadata} from "../core/Public/Models/Interfaces/ITweetWithSearchMetadata";
import {ITweetWithSearchMetadataDTO} from "../core/Public/Models/Interfaces/DTO/ITweetWithSearchMetadataDTO";
import {TweetMode} from "../core/Public/Settings/SharebookSettings";
import {ITwitterClient} from "../core/Public/ITwitterClient";
import {ITweetFromSearchMetadata} from '../core/Public/Models/Interfaces/DTO/ITweetFromSearchMetadata';

export class TweetWithSearchMetadata extends Tweet implements ITweetWithSearchMetadata {
  private readonly _tweetWithSearchMetadataDTO: ITweetWithSearchMetadataDTO;

  constructor(tweetDTO: ITweetWithSearchMetadataDTO, tweetMode?: TweetMode, client: ITwitterClient) {
    super(tweetDTO, tweetMode, client);
    this._tweetWithSearchMetadataDTO = tweetDTO;
  }

  get searchMetadata(): ITweetFromSearchMetadata {
    return this._tweetWithSearchMetadataDTO.tweetFromSearchMetadata;
  }
}
